### PR TITLE
[media-library] Add unit tests to module

### PR DESCRIPTION
**Pull request by barthap**
# Why
MediaLibrary has many utility methods, which behavior is sometimes unclear. Unit tests would describe that behavior.

Also, contains some minor fixes.

# How

 - Add native unit tests to `MediaLibrary` on Android.
 - Fixes bug, where `getAssetsAsync()` could return without promise rejection, when caught `IOException`
 - Adds a little input validation to module methods
 - Removes unused test dependencies from `expo-web-browser`. 

# Test plan

`et android-native-unit-tests`

---
WIP

- [x] Add testing dependencies
- [x] Add utility functions to mock `ContentResolver`
- [x] Test `GetQueryInfo`
- [x] Test `MediaLibraryUtils` where possible
- [x] Test `GetAssets`
- [x] Test `GetAssetInfo`
- [x] Extract useful test utility functions to `unimodules-test-core`
- [x] Input validation and minor refactors

### CODE REVIEW

File: packages/expo-media-library/android/src/test/java/expo/modules/medialibrary/MockData.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  )
+}
+
+internal object MockData {
+  val mockImage = MockAsset(
+    id = 1L,
+    name = "img1.jpg",
+    path = "images/img1.jpg",
+    mediaType = MEDIA_TYPE_IMAGE,
+    width = 100, height = 200,
```
Comment (lukmccall) [inline packages/expo-media-library/android/src/test/java/expo/modules/medialibrary/MockData.kt:39]:
```suggestion
    width = 100, 
    height = 200,
```
I know that if you put it in one line, it looks good 😅  But I don't like it 😜

File: packages/expo-media-library/android/src/test/java/expo/modules/medialibrary/MockData.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    duration = null,
+    albumId = "album12"
+  )
+
+  val mockVideo = MockAsset(
+    id = 2L,
+    name = "vid2.mp4",
+    path = "videos/vid2.mp4",
+    mediaType = MEDIA_TYPE_VIDEO,
+    width = 200, height = 300,
```
Comment (lukmccall) [inline packages/expo-media-library/android/src/test/java/expo/modules/medialibrary/MockData.kt:52]:
```suggestion
    width = 200,
    height = 300,
```
Same here

File: packages/expo-media-library/android/src/test/java/expo/modules/medialibrary/MockData.kt [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    duration = 3000,
+    albumId = "album23"
+  )
+
+  val mockAudio = MockAsset(
+    id = 3L,
+    name = "song3.mp3",
+    path = "audio/song3.mp3",
+    mediaType = MEDIA_TYPE_AUDIO,
+    width = 0, height = 0,
```
Comment (lukmccall) [inline packages/expo-media-library/android/src/test/java/expo/modules/medialibrary/MockData.kt:65]:
```suggestion
    width = 0, 
    height = 0,
```
and here

File: packages/expo-media-library/android/src/test/java/expo/modules/medialibrary/TestUtils.kt [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  every { contentResolver.query(
+    MediaLibraryConstants.EXTERNAL_CONTENT,
+    MediaLibraryConstants.ASSET_PROJECTION,
+    any(), any(), any())
+  } returns cursor
+  return contentResolver
+}
+
+fun mockContentResolverForResult(cursorResults: CursorResults)
+  = mockContentResolver(mockCursor(cursorResults))
```
Comment (lukmccall) [inline packages/expo-media-library/android/src/test/java/expo/modules/medialibrary/TestUtils.kt:30]: I think that the android studio will put it in one line 🤔

File: packages/expo-media-library/src/MediaLibrary.ts [hunk 5]
```diff
@@ -375,6 +375,13 @@ export async function getAssetsAsync(assetsOptions: AssetsOptions = {}): Promise
     throw new Error('Option "album" must be a string!');
   }
 
+  if (Platform.OS === 'android' && isNaN(parseInt(getId(after) as string, 10))) {
+    throw new Error('Option "after" must be a valid ID!');
+  }
+  if (first != null && first < 0) {
```
Comment (lukmccall) [inline packages/expo-media-library/src/MediaLibrary.ts:381]:
```suggestion

  if (first != null && first < 0) {
```

**Review by lukmccall — APPROVED** [review]
LGTM 💪

File: packages/expo-media-library/android/src/test/java/expo/modules/medialibrary/TestUtils.kt (hunk 4 above)
Comment (barthap) [reply packages/expo-media-library/android/src/test/java/expo/modules/medialibrary/TestUtils.kt:30]: If the line is too long, it automatically splits it to new line.

**Review by bbarthec — APPROVED** [review]
LGTM 👍

**Pull request merged**